openedx/core/djangoapps/profile_images/images.py

```diff
@@ -127,51 +135,104 @@ def validate_uploaded_image(uploaded_file):
     uploaded_file.seek(0)
 
 
-def _get_scaled_image_file(image_obj, size):
+def _crop_image_to_square(image):
+    """
+    Given a PIL.Image object, return a copy cropped to a square around the
+    center point with each side set to the size of the smaller dimension.
+    """
+    width, height = image.size
+    if width != height:
+        side = width if width < height else height
+        left = (width - side) // 2
+        top = (height - side) // 2
+        right = (width + side) // 2
+        bottom = (height + side) // 2
+        image = image.crop((left, top, right, bottom))
+    return image
+
+
+def _set_color_mode_to_rgb(image):
     """
-    Given a PIL.Image object, get a resized copy using `size` (square) and
-    return a file-like object containing the data saved as a JPEG.
+    Given a PIL.Image object, return a copy with the color mode set to RGB.
+    """
+    return image.convert('RGB')
+
+
+def _scale_image(image, side_length):
+    """
+    Given a PIL.Image object, get a resized copy with each side being
+    `side_length` pixels long.  The scaled image will always be square.
+    """
+    return image.resize((side_length, side_length), Image.ANTIALIAS)
+
 
-    Note that the file object returned is a django ContentFile which holds
-    data in memory (not on disk).
+def _create_image_file(image, exif):
+    """
+    Given a PIL.Image object, and return a file-like object containing the data
+    saved as a JPEG.
+
+    Note that the file object returned is a django ContentFile which holds data
+    in memory (not on disk).
     """
-    if image_obj.mode != "RGB":
-        image_obj = image_obj.convert("RGB")
-    scaled = image_obj.resize((size, size), Image.ANTIALIAS)
     string_io = StringIO()
-    scaled.save(string_io, format='JPEG')
+    if exif is None:
+        image.save(string_io, format='JPEG')
+    else:
```

Comment:
Yes.  If you pass exif=None to save, it breaks trying to perform operations on None.